services/communitytokens/service.go

```diff
@@ -68,14 +90,181 @@ func (s *Service) APIs() []ethRpc.API {
 
 // Start is run when a service is started.
 func (s *Service) Start() error {
+
+	s.walletWatcher = walletevent.NewWatcher(s.walletFeed, s.handleWalletEvent)
+	s.walletWatcher.Start()
+
 	return nil
 }
 
+func (s *Service) handleWalletEvent(event walletevent.Event) {
```

Comment:
This is the main change in the pr.
Here all events are handled and signal is emitted.